src/App.tsx

```diff
@@ -75,19 +72,7 @@ const App = () => {
     isLocked,
   } = useSelector((state: RootState) => state.app)
 
-  const { walletLinkToken } = useSelector((state: RootState) => state.app)
-  const [heliumWallet, setHeliumWallet] = useState<string>()
-
-  useEffect(() => {
-    if (!walletLinkToken) {
-      if (heliumWallet) {
-        setHeliumWallet('')
-      }
-      return
-    }
-
-    getAddress().then(setHeliumWallet)
-  }, [heliumWallet, walletLinkToken])
+  const { result: heliumWallet } = useAsync(getAddress, [])
```

Comment:
So if they are a new account and dont have a wallet it will just be undefined and go thought the link flow? Another thought, do we need to hide the reveal words and export key menu options when they dont have any secure store data?